modules/tensor_mechanics/src/materials/ADComputeFiniteStrain.C

```diff
@@ -147,9 +147,6 @@ ADComputeFiniteStrain::computeQpIncrements(ADRankTwoTensor & total_strain_increm
       const ADReal C1_squared =
           p + 3.0 * Utility::pow<2>(p) * (1.0 - (p + q)) / Utility::pow<2>(p + q) -
           2.0 * Utility::pow<3>(p) * (1.0 - (p + q)) / Utility::pow<3>(p + q);
-      mooseAssert(C1_squared >= 0.0,
-                  "Cannot take square root of a negative number. This may happen when elements "
-                  "become heavily distorted.");
       const ADReal C1 = std::sqrt(C1_squared);
```

Comment:
you just want to eat the NaN here?

Reply:
I do

Reply:
If C1_squared is zero - won't we get a floating point exception here due to AD?

Reply:
Yes sir. Or if it is negative. The difference is that you can disable floating point exceptions from the command line but you cannot disable asserts. So by default a user's simulation in dbg mode will abort at this exact same location, but an advanced user can choose to disable it from the command line if they wish

Reply:
I guess you could argue that the assertion message is more helpful than a backtrace

Reply:
And if you feel that way I'll close this PR and then go limit that blackbear test to run in non-debugging modes

Reply:
You know I think *I* like the assertion message better than the back trace, so I'm going to close